### DISCUSSION TITLE
MCCGQ02HL sensor support

### DISCUSSION THREAD

**Discussion by maxxer** [theengs/decoder#126] [Q&A]
Hi.
Is there any plan to support the sensor of the [Mijia Smart Door & Window Sensor 2](https://it.aliexpress.com/item/1005001376080696.html)? I see from the box the sensor seems to be MCCGQ02HL. 

Thanks

**Comment by DigiH** [comment]
Hi @maxxer, from the little information I found about the MCCGQ02HL is that it needs a connection for subscribing to a notification characteristic. Currently Theengs Gateway 

>  focuses on ble broadcaster not connection.

https://github.com/theengs/gateway/issues/26

If you have this Mijia Smart Door & Window Sensor 2 and do see broadcast messages (manufacturerdata/servicedata) when its contact is closed/opened however please let us know.

Thanks

**Reply by 1technophile** [reply]
I think this device broadcast also but the data are encrypted. So we would need to take into account a bind key and decrypt the messages.

**Reply by maxxer** [reply]
I managed to retrieve the keys from my Xiaomi account, is there any way I can use the device with the current code? thanks

**Reply by 1technophile** [reply]
not with the current code, we would need to add decryption of data leveraging the keys

**Reply by DigiH** [reply]
Read up a bit more about the new Xiaomi 2 sensors (door/window, motion/light, Water leak …), which look very similar to the previous Xiaomi Mi/Aqara devices, but all have moved away from zigbee to now use BT5.0. Interesting development.

**Reply by h2zero** [reply]
@DigiH when you said they moved to BT5.0, are they using the BT5.0 (extended) advertising? Interesting that they moved from zigbee to BT, I would have expected a move to Thread, I guess that's still too bleeding edge though.

**Reply by DigiH** [reply]
> when you said they moved to BT5.0, are they using the BT5.0 (extended) advertising?

That I don't know and didn't really investigate further. Only that pretty much the same looking device as my zigbee Xiaomi/Aqara door/window, motion/light, water leak sensors are now BT instead of zigbee. If this is also a push to sell the new Xiaomi/Aqara/Mi gateways which now also include BT on top of zigbee is anybody's guess ;)

Since most users talking about this mention that the keys are easy to obtain I was wondering if we could include a decrypt functionality at some stage with tinycrypt or such. 

> I would have expected a move to Thread

I would expect this to pick up later this year, and even if I already have six Thread border router compatible devices here, they are currently no real use to me.

With Xiaomi/Aqara/Mi they might wait a bit, then bring out a new gateway which also supports Thread, or possibly they already have one.

**Reply by h2zero** [reply]
>If this is also a push to sell the new Xiaomi/Aqara/Mi gateways which now also include BT on top of zigbee is anybody's guess ;)

High probability for sure.

>Since most users talking about this mention that the keys are easy to obtain I was wondering if we could include a decrypt functionality at some stage with tinycrypt or such.

I'm not sure how practical that would be. Could be doable if the device is detectable without having to decrypt the data first.

>I would expect this to pick up later this year, and even if I already have six Thread border router compatible devices here, they are currently no real use to me.

Which ones?

**Reply by DigiH** [reply]
> I'm not sure how practical that would be. Could be doable if the device is detectable without having to decrypt the data first.

AFAIK ble monitor uses the MAC address plus the linked key to then pick up the broadcasts and decrypt for further decoding. I suppose something similar would be needed if we'd want to include this functionality in the future. With Decoder being used by several backends though this would make it more complicated in where to enter such combinations on OMG, Theengs Gateway ...

> Which ones?

Apple TV and HomePod minis, I think among the few Thread border routers around so far, is the Nanoleaf out yet? I'm sure this will all change soon. In regards to Thread they are just sitting idle at the moment, but once it picks up I suppose I'll have a good tight mesh ;)

**Reply by h2zero** [reply]
>AFAIK ble monitor uses the MAC address plus the linked key to then pick up the broadcasts and decrypt for further decoding. I suppose something similar would be needed if we'd want to include this functionality in the future. With Decoder being used by several backends though this would make it more complicated in where to enter such combinations on OMG, Theengs Gateway 

Could be problematic for sure, not really within the scope of this library. It would need to have the data decrypted before decoding.

>Apple TV and HomePod minis, I think among the few Thread border routers around so far, is the Nanoleaf out yet?

Cool, I want to create my own border router and devices. I have lots of nordic boards around to work with and it would be helpful to test the Thread implementation with some commercial stuff.

**Reply by DigiH** [reply]
> Cool, I want to create my own border router and devices. I have lots of nordic boards around to work with and it would be helpful to test the Thread implementation with some commercial stuff.

Sounds exciting, Eve, Nanoleaf and likely several others for sensors/devices now - I think I very recently read something about Ikea Tradfri kicking off as well - but commercial border routers pretty sparse still. Will be interesting to hear how you get on with it all :)

**Reply by h2zero** [reply]
It's all very early stages right now but I think Matter + Thread will be exciting in the future.

**Reply by DigiH** [reply]
I think so too, and it's really about time to get some good, and more importantly working, convergence and interoperability!

While half of it is actually fun and what we like doing here ;) the other half can be so frustrating with many different proprietary protocols.

**Reply by DigiH** [reply]
Signing off here for the night. Have a good one, and thanks again for the BM2 fix 👍
